tests/test_callbacks.cpp

```diff
@@ -14,6 +14,32 @@
 
 #include <thread>
 
+namespace test_callbacks {
+namespace boost_histogram { // See PR #5580
+
+double custom_transform_double(double value) { return value * 3; }
+int custom_transform_int(int value) { return value; }
+
+// Originally derived from
+// https://github.com/scikit-hep/boost-histogram/blob/460ef90905d6a8a9e6dd3beddfe7b4b49b364579/include/bh_python/transform.hpp#L68-L85
+double apply_custom_transform(const py::object &src, double value) {
+    using raw_t = double(double);
+
+    py::detail::make_caster<std::function<raw_t>> func_caster;
+    if (!func_caster.load(src, /*convert*/ false)) {
+        return -100;
+    }
+    auto func = static_cast<std::function<raw_t> &>(func_caster);
+    auto cfunc = func.target<raw_t *>();
```

Comment:
```suggestion
    raw_t* cfunc = func.target<raw_t *>();
```

Reply:
Nevermind, you fixed it

Reply:
Oh, I just pusehd the `auto *` change as suggested by clang-tiday.